_playwright-tests/UI/GPG_keys.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { test, expect } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist, getRowByNameOrUrl } from './helpers/helpers';
+
+export const repoNamePrefix = 'GPG-key';
+export const randomName = () => (Math.random() + 1).toString(36).substring(2, 6);
+export const repoName = `${repoNamePrefix}-${randomName()}`;
+
+export const url = 'https://jlsherrill.fedorapeople.org/fake-repos/signed/';
+export const packages_key =
+  'http://jlsherrill.fedorapeople.org/fake-repos/needed-errata/RPM-GPG-KEY-dummy-packages-generator';
+export const meta_key = 'https://jlsherrill.fedorapeople.org/fake-repos/signed/GPG-KEY.gpg';
+
+test.describe('Test GPG keys', () => {
+  test('Create a repo, add a GPG Key, toggle metadata', async ({ page }) => {
+    await test.step('Delete any GPG key test repos that exist', async () => {
+      await navigateToRepositories(page);
+      await closePopupsIfExist(page);
+      await deleteAllRepos(page, `&search=${repoNamePrefix}`);
+    });
+    await test.step('Create a repository', async () => {
+      // Click on the 'Add repositories' button
+      // HMS-5268 There are two buttons on the ZeroState page
+      await page.getByRole('button', { name: 'Add repositories' }).first().click();
+      await expect(page.getByRole('dialog', { name: 'Add custom repositories' })).toBeVisible();
+      // Fill in the repository details
+      await page.getByLabel('Name').fill(`${repoName}`);
+      await page.getByLabel('Introspect only').click();
+      await page.getByLabel('URL').fill(url);
+      // Paste the URL to the packages GPG key
+      await page.getByPlaceholder('Paste GPG key or URL here').fill(packages_key);
```

Comment:
This test should somehow verify that it has the correct key. Can you verify it somewhere that it uses the correct key? Do we see it somewhere? Or can we verify that we cannot access something without it and once we add it we can? Or something similar?
I wonder what is the value of this test - is it that we can enter value into this field or that if you enter the value you can do something.

Reply:
If the key was not in a valid format the Save button would not be enabled. 
Our test just shows we can enter a key and it is checked for valid format and saved.

It is up to the user or clients to use, or not, the key as they see fit.

Reply:
I have add a check that validate fails if you select the metadata option for a package verification only key.